src/ray/core_worker/core_worker.cc

```diff
@@ -19,11 +19,16 @@ void BuildCommonTaskSpec(
   // Build common task spec.
   builder.SetCommonTaskSpec(task_id, function.GetLanguage(),
                             function.GetFunctionDescriptor(), job_id, current_task_id,
-                            task_index, caller_id, num_returns, required_resources,
-                            required_placement_resources);
+                            task_index, caller_id, num_returns,
+                            transport_type == ray::TaskTransportType::DIRECT,
+                            required_resources, required_placement_resources);
   // Set task arguments.
   for (const auto &arg : args) {
     if (arg.IsPassedByReference()) {
+      if (transport_type == ray::TaskTransportType::RAYLET) {
+        RAY_CHECK(!arg.GetReference().IsDirectCallType())
+            << "NotImplemented: passing direct call objects to other tasks";
```

Comment:
better return an error instead of crashing the worker?

Reply:
also the error message should be "to other non-direct tasks"?

Reply:
I'm fine with crashing Ray since this is an experimental feature.

Reply:
ok, could you fix the error message?